cylc/flow/pathutil.py

```diff
@@ -295,6 +295,40 @@ def remove_dir_or_file(path: Union[Path, str]) -> None:
         rmtree(path, onerror=handle_rmtree_err)
 
 
+def remove_empty_parents(
+    path: Union[Path, str], tail: Union[Path, str]
+) -> None:
+    """Work our way up the tail of path, removing empty dirs only.
+
+    Args:
+        path: Absolute path to the directory, e.g. /foo/bar/a/b/c
+        tail: The tail of the path to work our way up, e.g. a/b/c
+
+    Example:
+        remove_empty_parents('/foo/bar/a/b/c', 'a/b/c') would remove
+        /foo/bar/a/b (assuming it's empty), then /foo/bar/a (assuming it's
+        empty).
+    """
+    path = Path(path)
+    if not path.is_absolute():
+        raise ValueError('path must be absolute')
```

Comment:
Would be good to cover some of these exception lines.

Reply:
Codecov is lying 😢 https://github.com/cylc/cylc-flow/pull/4289/files#diff-a9a7ea82148a0b1c8f631db722616f6cca1f468f3bf77bf83cd783a93220c7edR439-R442

Reply:
Yep, confirmed locally, went into the codecov report and found that the problem doesn't exist there. Will try fiddling the codecov config as I think these comments may have been left *before* all jobs had finished.

Reply:
@oliver-sanders See #4348